File: Dev/C++/Mascaret/src/VEHA/Entity/AnimationSpecification.cpp
#include "VEHA/Entity/AnimationSpecification.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "MascaretApplication.h"
#include "VEHA/Kernel/Model.h"

namespace VEHA
{

AnimationSpecification::AnimationSpecification(const string & name, const string & url)
:VirtualRealitySpecification(MascaretApplication::getInstance()->getModel()->getBasicType("animation")), _name(name), _url(url)
{
}

AnimationSpecification::~AnimationSpecification()
{
}

shared_ptr<ValueSpecification> AnimationSpecification::clone() const
{
	throw std::exception();
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Entity/AnimationSpecification.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public abstract class AnimationSpecification : VirtualRealitySpecification
    {
        public string name;

        private string url;
        public string Url
        {
            get { return url; }
            set { url = value; }
        }

        //callbacks missing

        public AnimationSpecification(string name, string url)
            : base(MascaretApplication.Instance.Model.getBasicType("animation"))
        {
            this.name = name;
            this.url = url;
        }

        public override ValueSpecification clone()
        {
            throw new NotImplementedException();
        }

        //callbakcs operation

        public abstract void setShape(ShapeSpecification shape);
        public abstract ShapeSpecification getShape();
        //Default parameters : speed = 1.0 sens = 1 cycle = false
        public abstract bool play(double speed, int sens, bool cycle);
        public abstract bool stop();
    }
}


